docs/watch-filtering/README.md

```diff
@@ -0,0 +1,179 @@
+streams, to receive continuous updates from the server about resources it
+manages. These list and watch requests are generally already more efficient than
+making many get requests for every individual object, but they trade higher
+memory use for fewer API calls and lower latency. In addition to the CPU and
+memory used to process these API calls, caching is used to detect drift,
+tolerate missed events, and reduce the need to make asynchronous get calls.
+
+Historically, Config Sync has used namespace filtering to reduce the scope of
+list and watch calls, reducing the compute needed to process and cache the
+results. However, in version 1.19.0, we also added label filtering, which
+further reduces the number of objects processed by Config Sync, down to just the
+ones that have the labels used by that specific reconciler.
+
+With this change, reconcilers will no longer receive list entries or watch
+events for objects they don't manage, and those objects will not be cached in
+the reconciler's memory. This reduces the memory used by reconcilers that have
+overlapping resource types and objects in the same namespaces.
+
+### Affected Use Cases
+
+-  RootSyncs that manage objects with the same resource type.
+-  RepoSync that manage objects with the same resource type in the same namespace.
+-  RootSyncs and RepoSync that manage objects with the same resource type in the
+   same namespace.
+
+Exception: RootSync reconcilers have an additional optimization which reduces
+the scope of watches when all managed objects are in the same namespace. This
+makes them perform more like a RepoSync reconciler is this particular scenario,
+which reduces the effectiveness of label filtering relative to other RootSync
+reconcilers, which watch resources in all namespaces.
+
+### Scenario A: 10 RootSyncs with 400 Deployments Each
+
+In this scenario, 10 RootSyncs are created, each pointing to a different
+directory in the same Git repository. Each directory contains a number of
+uniquely named Deployment manifests, all in the same namespace.
+
+This demonstrates a best-case scenario for watch filtering where all the managed
+objects are large, the same resource type, and in the same namespace.
+
+![Scenario A Results](scenario-a-results.png)
+
+These results show a linear increase in memory reduction, as the number of
+overlapping resources increase.
+
+### Scenario B: 1 RootSync with 2000 Deployments Each
+
+In this scenario, 1 RootSync is created, pointing to a Git repository containing
+a number of uniquely named Deployment manifests, all in the same namespace.
+
+This demonstrates a worst-case scenario for watch filtering where none of the
+resources being watched overlap with any other objects on the cluster.
+
+![Scenario B Results](scenario-b-results.png)
+
+These results show no statistically relevant change in memory use between
+Config Sync versions 1.18 and 1.19.
+
+So if you only ever use one RootSync, you're unlikely to see much reduction in
```

Comment:
What if there are other objects of the watched GVK which are not managed by Config Sync? Were we already using label filtering that filtered out objects not managed by CS? I don't believe we were, so might be worth noting that

Reply:
I mentioned it elsewhere, but I'll add a caveat here.